alacritty/src/renderer/box_drawing.rs

```diff
@@ -0,0 +1,584 @@
+                    (canvas.center_h(), canvas.center_h(), width)
+                },
+                '\u{2552}'..='\u{2554}' | '\u{2568}' => (v_left_bounds.0, v_right_bounds.0, width),
+                '\u{2558}'..='\u{255a}' => (v_right_bounds.0, v_left_bounds.0, width),
+                '\u{255e}'..='\u{2560}' | '\u{256a}' | '\u{256c}' => {
+                    (v_right_bounds.0, v_right_bounds.0, width)
+                },
+                '\u{2564}' | '\u{2566}' => (canvas.center_h(), v_right_bounds.0, width),
+                '\u{2567}' | '\u{2569}' => (v_right_bounds.0, canvas.center_h(), width),
+                _ => (0., 0., 0.),
+            };
+
+            // Top vertical part.
+            let (left_top_size, right_top_size) = match character {
+                '\u{2551}' | '\u{256a}' => (canvas.center_v(), canvas.center_v()),
+                '\u{2558}'..='\u{255c}' | '\u{2567}' | '\u{2568}' => {
+                    (h_bot_bounds.1, h_top_bounds.1)
+                },
+                '\u{255d}' => (h_top_bounds.1, h_bot_bounds.1),
+                '\u{255e}'..='\u{2560}' => (canvas.center_v(), h_top_bounds.1),
+                '\u{2561}'..='\u{2563}' => (h_top_bounds.1, canvas.center_v()),
+                '\u{2569}' | '\u{256b}' | '\u{256c}' => (h_top_bounds.1, h_top_bounds.1),
+                _ => (0., 0.),
+            };
+
+            // Bottom vertical part.
+            let (left_bot_y, right_bot_y, bottom_size) = match character {
+                '\u{2551}' | '\u{256a}' => (canvas.center_v(), canvas.center_v(), height),
+                '\u{2552}'..='\u{2554}' => (h_top_bounds.0, h_bot_bounds.0, height),
+                '\u{2555}'..='\u{2557}' => (h_bot_bounds.0, h_top_bounds.0, height),
+                '\u{255e}'..='\u{2560}' => (canvas.center_v(), h_bot_bounds.0, height),
+                '\u{2561}'..='\u{2563}' => (h_bot_bounds.0, canvas.center_v(), height),
+                '\u{2564}'..='\u{2566}' | '\u{256b}' | '\u{256c}' => {
+                    (h_bot_bounds.0, h_bot_bounds.0, height)
+                },
+                _ => (0., 0., 0.),
+            };
+
+            // Left horizontal line.
+            canvas.draw_h_line(0., h_lines.0, top_left_size, stroke_size);
+            canvas.draw_h_line(0., h_lines.1, bot_left_size, stroke_size);
+
+            // Right horizontal line.
+            canvas.draw_h_line(top_right_x, h_lines.0, right_size, stroke_size);
+            canvas.draw_h_line(bot_right_x, h_lines.1, right_size, stroke_size);
+
+            // Top vertical line.
+            canvas.draw_v_line(v_lines.0, 0., left_top_size, stroke_size);
+            canvas.draw_v_line(v_lines.1, 0., right_top_size, stroke_size);
+
+            // Bottom vertical line.
+            canvas.draw_v_line(v_lines.0, left_bot_y, bottom_size, stroke_size);
+            canvas.draw_v_line(v_lines.1, right_bot_y, bottom_size, stroke_size);
+        },
+        // Arcs: '╭', '╮', '╯', '╰'.
```

Comment:
I wasn't even sure if I should mention this, but there might be something like a one-off caused by flipping around the arc in edge-cases?

At least when I combine them with the light horizontal dash, the top and bottom look slightly different:
![tmp](https://user-images.githubusercontent.com/8886672/147889558-454ada4e-21c0-426c-9f0e-9f67ded7eb20.png)

Notice how at the bottom the thing is completely flat even into the arc, while at the top it seems to step down more aggressively and sooner? I'm pretty sure these two are supposed to be used together (arcs and light horizontal/vertical), so it would probably make sense to ensure they meet up properly.